containerfiles/Dockerfile

```diff
@@ -90,13 +66,11 @@ RUN \
       TARGET_TRIPLE=$(uname -m)-unknown-linux-gnu; \
     fi; \
     export PROTOC=/usr/local/bin/protoc; \
-    cargo build --release --target $TARGET_TRIPLE --bin $TARGETBINARY;
-
-# replace this with `--out` or `--out-dir` once stable
-RUN mkdir -p target/release
-RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then ARCH=aarch64; \
+    cargo build --release --target $TARGET_TRIPLE --bin $TARGETBINARY;\
+    # Copies the binary from out of the cache directory
+    if [ "$TARGETPLATFORM" = "linux/arm64" ]; then ARCH=aarch64; \
     elif [ "$TARGETPLATFORM" = "linux/amd64" ]; then ARCH=x86_64; fi; \
-    cp target/$ARCH-unknown-linux-gnu/release/$TARGETBINARY target/release/
+    cp target/$ARCH-unknown-linux-gnu/release/$TARGETBINARY release/;
```

Comment:
What's the reason for doing this? Is moving from `target/$ARCH-unknown-linux-gnu/release` to `target/release` invalidating the cache?

Reply:
the cache is all of target/, if you run in a different command the file doesn't exist because you need to mount the cache. By moving the artifact out of the cache it persists.